Tests/test_reader.py

```diff
@@ -214,16 +214,11 @@ def test_get_images_raw(strict, with_prev_0, should_fail):
         PdfFileReader(pdf_stream, strict=strict)
 
 
-@pytest.mark.xfail(
-    reason=(
-        "It's still broken - and unclear what the issue is. "
-        "Help would be appreciated!"
-    )
-)
 def test_issue297():
     path = os.path.join(RESOURCE_ROOT, "issue-297.pdf")
-    reader = PdfFileReader(path, "rb")
-    reader.getPage(0)
+    with pytest.raises(UserWarning):
+        reader = PdfFileReader(path, "rb")
+        reader.getPage(0)
```

Comment:
This deactivates the test. At least for `strict=False` I would expect `reader.extractText()` to give something reasonable.

The rest of the PR might be fine, but it does not fix issue297. I will not merge this part.